controller-server/src/main/java/com/yahoo/vespa/hosted/controller/auditlog/AuditLog.java

```diff
@@ -112,16 +110,27 @@ public enum Method {
             DELETE
         }
 
-        private static Optional<String> sanitize(Optional<String> data) {
-            Objects.requireNonNull(data, "data must be non-null");
-            return data.filter(Predicate.not(String::isBlank))
-                       .filter(CharMatcher.ascii()::matchesAllOf)
-                       .map(v -> {
-                           if (v.length() > maxDataLength) {
-                               return v.substring(0, maxDataLength);
-                           }
-                           return v;
-                       });
+        private static Optional<String> sanitize(byte[] data) {
+            StringBuilder sb = new StringBuilder();
+            for (byte b : data) {
+                char c = (char) b;
+                if (!printableAscii(c) && !tabOrLineBreak(c)) {
+                    break;
```

Comment:
Return `Optional.empty()` here? Don't want to return ASCII prefix of a binary file

Reply:
PTAL